pkg/synopsysctl/utils.go

```diff
@@ -212,3 +215,42 @@ func RunKubeEditorCmd(restConfig *rest.Config, kube bool, openshift bool, args .
 	//time.Sleep(1 * time.Second) TODO why did Jay put this here???
 	return nil
 }
+
+func ctlUpdateResource(resource interface{}, mock bool, mockFormat string, kubeMock bool, mockKubeFormat string) error {
+	if mock {
+		log.Debugf("running mock mode")
+		err := PrintResource(resource, mockFormat, false)
+		if err != nil {
+			return fmt.Errorf("failed to print resource: %s", err)
+		}
+	} else if kubeMock {
+		log.Debugf("running kube mock mode")
+		err := PrintResource(resource, mockKubeFormat, true)
+		if err != nil {
+			return fmt.Errorf("failed to print resource: %s", err)
+		}
+	} else {
+		if alt, ok := resource.(*alertapi.Alert); ok {
```

Comment:
Use reflect.TypeOf to make this a switch statement.

Reply:
@rrati I am using the built in .(type) , should I change it to use reflect.TypeOf()?  Do you have an example?

Reply:
https://github.com/koki/mantle/blob/master/pkg/core/pod/from_kube.go#L17

Reply:
Done